Realm/RLMArrayLinkView.mm

```diff
@@ -79,6 +79,10 @@ - (NSUInteger)count {
     return _backingLinkView->size();
 }
 
+- (BOOL)isInvalidated {
+    return !_backingLinkView->is_attached() || !RLMMatchesThread(_realm);
```

Comment:
Checking if it's on the correct thread is inconsistent with RLMObject and is conflating two completely different things.